src/main/java/sg/dex/starfish/keeper/DirectPurchase.java

```diff
@@ -0,0 +1,189 @@
+package sg.dex.starfish.keeper;
```

Comment:
Scope: OK with this package in starfish for now, but it really feels like this is getting too deep into the specifics of the Ocean Network and should be moved to Squid or a separate artifact in the longer term.

Might be worth thinking about creating a `starfish-ocean` module which includes this and the squid implementation. @AyushVerma2 have you given any more thought to this?